test/quantization/pt2e/test_quantize_pt2e.py

```diff
@@ -1572,6 +1585,131 @@ def test_qat_conv_bn_relu_numerics(self):
             m, example_inputs, is_per_channel=True, verify_convert=True,
         )
 
+    def test_q_dq_decomposition(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.conv = torch.nn.Conv2d(1, 1, 1)
+
+            def forward(self, x):
+                x = self.conv(x)
+                return x
+
+        with override_quantized_engine("qnnpack"):
+            m = M().eval()
+            example_inputs = (torch.randn(1, 1, 3, 3),)
+
+            # program capture
+            m, guards = torchdynamo.export(
+                m,
+                *copy.deepcopy(example_inputs),
+                aten_graph=True,
+                tracing_mode="real",
+            )
+
+            quantizer = QNNPackQuantizer()
+            quantizer.set_global(
+                get_symmetric_quantization_config(
+                    is_per_channel=False, is_qat=False
+                )
+            )
+            m = prepare_pt2e_quantizer(m, quantizer)
+            m(*example_inputs)
+            m = convert_pt2e(m)
+            m(*example_inputs)
+            node_occurrence = {
+                # two for input and weight of the conv, one for output for the conv
+                ns.call_function(torch.ops.quantized_decomposed.quantize_per_tensor.default): 3,
+                ns.call_function(torch.ops.quantized_decomposed.dequantize_per_tensor.default): 3,
+            }
+            node_list = [
+                ns.call_function(torch.ops.quantized_decomposed.dequantize_per_tensor.default),
+                ns.call_function(torch.ops.quantized_decomposed.dequantize_per_tensor.default),
+                ns.call_function(torch.ops.aten.convolution.default),
+                ns.call_function(torch.ops.quantized_decomposed.dequantize_per_tensor.default),
+            ]
+            self.checkGraphModuleNodes(
+                m,
+                expected_node_list=node_list,
+                expected_node_occurrence=node_occurrence
+            )
+            m = make_fx(m, decomposition_table=quant_decomp)(*copy.deepcopy(example_inputs))
```

Comment:
So we should ahve the decomposition, right out of convert, hence my comment around not needing to register decomps for q/dq. This is something quant workflow needs to handle.

Reply:
makes sense, I'll rewrite this to be a rewrite instead of a decomposition